contracts/BaseLaunchpeg.sol

```diff
@@ -367,6 +379,36 @@ abstract contract BaseLaunchpeg is
         );
     }
 
+    /// @notice Set the public sale end time. Can only be set after phases
+    /// have been initialized. Cannot be set once public sale has started.
+    /// @dev Only callable by owner
+    /// @param _publicSaleEndTime new public sale end time
+    function setPublicSaleEndTime(uint256 _publicSaleEndTime)
+        external
+        override
+        onlyOwner
+    {
+        _setPublicSaleEndTime(_publicSaleEndTime);
+    }
+
+    /// @notice Set the public sale end time. Can only be set after phases
+    /// have been initialized. Cannot be set once public sale has started.
+    /// @param _publicSaleEndTime new public sale end time
+    function _setPublicSaleEndTime(uint256 _publicSaleEndTime) internal {
+        if (publicSaleEndTime == 0) {
+            revert Launchpeg__NotInitialized();
+        }
+        if (block.timestamp >= publicSaleStartTime) {
```

Comment:
do we want to allow updating of the public sale end time during the public sale phase as well (ie. after public sale start time)?

Reply:
Yeah, even if the sale is over, we could probably leave the possibility to reopen it by updating the `publicSaleEndTime`

Reply:
hm for this use case, we could use the "pause" function (that we don't have yet) ?
In my mind, ending the public sale means burning the remaining tokens forever but I might be wrong

Reply:
I was more thinking about the artist changing his mind and wanting to reopen the sale after closing it, or if we make a mistake on the end time and it happens before we notice.

I prefer having the most flexibility as possible, and we still have the possibility to burn ownership rights if we really want to properly close the sale